prov/sockets/src/sock_atomic.c

```diff
@@ -161,24 +163,41 @@ ssize_t sock_ep_tx_atomic(struct fid_ep *ep,
 	if (flags & FI_REMOTE_CQ_DATA)
 		sock_tx_ctx_write(tx_ctx, &msg->data, sizeof(uint64_t));
 
-	src_len = 0;
+	src_len = dst_len = 0;
 	if (flags & FI_INJECT) {
 		for (i = 0; i < msg->iov_count; i++) {
 			sock_tx_ctx_write(tx_ctx, msg->msg_iov[i].addr,
 					  msg->msg_iov[i].count * datatype_sz);
 			src_len += (msg->msg_iov[i].count * datatype_sz);
 		}
+		for (i = 0; i < compare_count; i++) {
```

Comment:
I have changed the processing part as well. I grouped the ringbuffer copy/read in the same place as otherwise another check is needed for compare data.